### PR TITLE
tools: Add descriptions to metrics help

### PR DESCRIPTION
**Pull request by grahamwhaley**
Add extended 'help' for the metrics, which comes out in the help on the commandline. We can thus expand details on what each metric measure does. The intention is also to add this information to the general repo markdown documents (so folks don't have to go reference the original PDF directly).

Also fix up some minor missing `local` vars.

### CODE REVIEW

**Comment by grahamwhaley** [comment]
If we ack these are going in the right direction, I'll expand them out into all the metrics (and fix the s/debug/verbose/ in the commit body :-) ).

File: tools/metric_itlb_mpki [hunk 1]
```diff
@@ -42,17 +46,21 @@ function init_itlb_mpki() {
 
 function calc_itlb_default_mpki() {
   local perf_data_file="$1"
+  local a=`return_pmu_value "itlb_misses.walk_completed" ${perf_data_file}`
+  local b=`return_pmu_value "instructions" ${perf_data_file}`
+
+  local verbose_a=`expand_if_verbose "[$a]"`
```
Comment (grahamwhaley) [inline tools/metric_itlb_mpki:52]: Ah, having now discovered the verbose option and what it already produced, such as:

```
 Performance counter stats for './run.sh':

    27,506,199,587      cycles:u
    19,794,238,994      instructions:u            #    0.72  insn per cycle
         2,564,841      itlb_misses.walk_completed:u
         2,564,841      itlb_misses.walk_completed_4k:u
                 0      itlb_misses.walk_completed_2m_4m:u
                 0      itlb_misses.walk_completed_1g:u

      60.006276325 seconds time elapsed

       9.375289000 seconds user
      24.967090000 seconds sys
```

these additions (of the counters post-pending the formulas) may be un-necessary.
What we do need is *documentation* though... to make it clear what features are available and how they can help diagnose runs.

Comment (uttampawar) [reply tools/metric_itlb_mpki:52]: @grahamwhaley How this is different than existing -v option?

Comment (uttampawar) [reply tools/metric_itlb_mpki:52]: @grahamwhaley  I like the idea of help function in each metric script which will be called from main measure-perf script. Should we expand to all the metric scripts?

Comment (grahamwhaley) [reply tools/metric_itlb_mpki:52]: > @grahamwhaley How this is different than existing -v option?

it's not - which I only found/realised later. As noted, I think we can drop the verbosity commit here. We need better docs.....

Comment (grahamwhaley) [reply tools/metric_itlb_mpki:52]: > @grahamwhaley I like the idea of help function in each metric script which will be called from main measure-perf script. Should we expand to all the metric scripts?

Yes, I would update for all the scripts, as noted in the PR message. I made this a simple one off and RFC to get feedback before investing the effort.

I'll drop the verbose commit and add the descriptions for the rest of the scripts and then PR an update for review. thanks!

**Comment by grahamwhaley** [comment]
I've updated - dropped the verbosity and added an expanded description for each metric.

I have a couple of questions whilst here:

1) In https://github.com/intel/iodlr/blob/master/tools/metric_l1_code_read_MPI, I see we:
 - use an l2 counter `l2_rqsts.all_code_rd`
 - multiply the math by 100 `100*(${a}/${b})`

    Are those intentional/correct? They don't seem to correlate with the name of the metric (l1 vs l2), or the implication of the data (MPI probably does not want the `100*`? - `metric_l2_demand_code_MPI` for instance afaict does not have the `*100`.

2) From the [`metric_icache_miss_stalls`](https://github.com/intel/iodlr/blob/master/tools/metric_icache_miss_stalls#L57), what is the counter `iicache_16b.ifdata_stall:c1:e1` - I've not found any references or descriptions for it?

**Comment by suresh-srinivas** [comment]
> I've updated - dropped the verbosity and added an expanded description for each metric.
> 
> I have a couple of questions whilst here:
> 
> 1. In https://github.com/intel/iodlr/blob/master/tools/metric_l1_code_read_MPI, I see we:
> 
> * use an l2 counter `l2_rqsts.all_code_rd`
> * multiply the math by 100 `100*(${a}/${b})`
>   Are those intentional/correct? They don't seem to correlate with the name of the metric (l1 vs l2),

Yes that is correct. We are calculating the L1 read miss and we are using the l2_rqsts_all to determine that. 

And yes we should not be multplying by 100. We an change this to MPKI (For Misses Per Kilo Instruction) and then multiplying by 1000 would make sense.

 or the implication of the data (MPI probably does not want the `100*`? - `metric_l2_demand_code_MPI` for instance afaict does not have the `*100`.
> 
> 1. From the [`metric_icache_miss_stalls`](https://github.com/intel/iodlr/blob/master/tools/metric_icache_miss_stalls#L57), what is the counter `iicache_16b.ifdata_stall:c1:e1` - I've not found any references or descriptions for it?

This is a named counter, see line 32. I needs the umask and cmask.
`"cpu/event=0x80,umask=0x4,cmask=1,edge=1,name=iicache_16b.ifdata_stall:c1:e1/"
`

**Comment by grahamwhaley** [comment]
Rebased and resolved the conflict @uttampawar - thx!

**Pull request merged**